src/Http/Routing/ref/Microsoft.AspNetCore.Routing.netcoreapp.cs

```diff
@@ -90,6 +90,36 @@ public partial class EndpointNameMetadata : Microsoft.AspNetCore.Routing.IEndpoi
     }
     [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Method, AllowMultiple=false, Inherited=false)]
     [System.Diagnostics.DebuggerDisplayAttribute("{DebuggerToString(),nq}")]
+    public sealed partial class HeaderAttribute : System.Attribute, Microsoft.AspNetCore.Routing.IHeaderMetadata
+    {
+        public HeaderAttribute(string headerName) { }
+        public HeaderAttribute(string headerName, string headerValue) { }
+        public HeaderAttribute(string headerName, string[] headerValues) { }
+        public string HeaderName { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } }
+        public Microsoft.AspNetCore.Routing.HeaderValueMatchMode HeaderValueMatchMode { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
+        public System.Collections.Generic.IReadOnlyList<string> HeaderValues { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } }
+        public System.StringComparison HeaderValueStringComparison { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
+    }
+    public sealed partial class HeaderMatcherPolicy : Microsoft.AspNetCore.Routing.MatcherPolicy, Microsoft.AspNetCore.Routing.Matching.IEndpointComparerPolicy, Microsoft.AspNetCore.Routing.Matching.IEndpointSelectorPolicy
```

Comment:
Can we make the matcherpolicy internal?

Reply:
We sure can. I modeled after `HostMatcherPolicy`. Was there a reason for that to be public?

Reply:
I'm not sure why we left the host matcher policy public, but normally we make all these implementations internal by default since that gives us more freedom to change them in the future.